web/src/popover_menus.js

```diff
@@ -276,16 +275,7 @@ export function register_popover_menu(target, popover_props) {
         e.stopPropagation();
         const instance = e.currentTarget._tippy;
 
-        if (instance) {
-            instance.hide();
-            return;
-        }
-
-        tippy(e.currentTarget, {
-            ...default_popover_props,
-            showOnCreate: true,
-            ...popover_props,
-        });
+        toggle_popover_menu(instance, e.currentTarget, popover_props);
```

Comment:
This commit would be better squashed into the earlier commit that introduces this function, so we don't have the messy intermediate version. Then it could have a "popover_menus: Extract toggle_popover_menu." commit message and clearly just extract a new function. And then it could be early in the branch and merged to unblock other PRs to use this without an intermediate state with duplicate code.

I also don't understand why it needs both the `target` parameter and te the `instance` parameter when the `instance` always equals `target._tippy`; will future callers do something different?

Reply:
I agree with both, so:
- I squashed the changes into a55d9020
- I removed the redundant instance parameter, so `toggle_popover_menu` calculates `instance` from `target`.

Reply:
Great!